FilecoinFoundationWeb/filecoin-foundation
# [UXIT-1507] Add ecosystem project submit form

## 📝 Description

This PR introduces a form for ecosystem project owners to submit their information for inclusion on our website. The form is available at `/ecosystem-explorer/project-form`

**New feature**

## 🛠️ Key Changes

- Adds a new page at `/ecosystem-explorer/project-form` using the generate page script.
- Creates a folder for ecosystem page content at `src/content/pages/ecosystem-explorer/*md`.
- Adds all form-related code (components, utils, constants, hooks, and schema) in `app/ecosystem-explorer/project-form`.
- Updates `ControlledFormFileInput.tsx` to accept and return `Array<File>` instead of `FileList`. This simplifies form initialization by allowing an empty array instead of using `new FileList`, which is not available server-side.
- Updates `ControlledFormFileInput.tsx` to accept `undefined` for optional values.
- Updates `FormLabel.tsx` to accept an `optional` prop, to display _(optional)_ next to the form label
- Removes the Airtable link from the `/ecosystem-explorer` page header and links it to the new form.

## 🧪 How to Test

1. Go to /ecosystem-explorer/project-form.
2. Fill out and submit the form.
3. Verify that the created PR contains all the submitted information in the correct format.
4. Open the preview link and confirm that the ecosystem project is displayed with the correct format.

## 📸 Screenshots

![CleanShot 2024-09-12 at 11 48 11@2x](https://github.com/user-attachments/assets/73bc007f-982f-477e-90ec-e263f8e3921b)
